### ISSUE TITLE
spring boot admin does not detect service after restarting spring boot admin

### ISSUE THREAD

**Issue by star4j** [codecentric/spring-boot-admin#1646]
spring boot 2.4.3
spring cloud 2020.0.1
spring boot admin 2.4.0

I have an environment with two services.
They are exchange-api-test and admin-test(is a spring-boot-admin project) 
They all are deployed and managed by Docker images and Kubernetes.
 I also use admin-test in order to monitor them all and spring-cloud-kubernetes to discover all the services automatically.

It works perfectly whenever I restart the exchange-api-test project (**kubectl rollout restart deployment exchange-api-deployment**), admin-test discovers every service.
![image](https://user-images.githubusercontent.com/25194485/109251114-e1a8ef00-7825-11eb-83de-4f0083d8e458.png)
 My problem comes when I restart admin-test project(**kubectl rollout restart deployment admin-deployment**), exchange-api-test is shown as OFFLINE in the SBA and it does not change its status.
![image](https://user-images.githubusercontent.com/25194485/109251424-6bf15300-7826-11eb-8377-d2f58475825c.png)
**applaction.yml (in admin-test project)**
> ```
>server:
>  port: 8080
>spring:
>  profiles:
>    active: test
>  application:
>    name: admin-${spring.profiles.active}
>  cloud:
>    kubernetes:
>      discovery:
>        all-namespaces: true
>        service-labels:
>          env: ${spring.profiles.active}
>      reload:
>        enabled: true
>management:
>  endpoints:
>    web:
>      exposure:
>        include: "*"
>  endpoint:
>    health:
>      show-details: always
> ```

**Comment by SteKoe** [comment]
Hi @star4j, have you added `@EnableScheduling` to the main Application class of your Spring Boot Admin implementation (see #1506)? 

See also:
https://cloud.spring.io/spring-cloud-kubernetes/1.1.x/reference/html/#secrets-propertysource

**Comment by star4j** [comment]
> Hi @star4j, have you added `@EnableScheduling` to the main Application class of your Spring Boot Admin implementation (see #1506)?
> 
> See also:
> https://cloud.spring.io/spring-cloud-kubernetes/1.1.x/reference/html/#secrets-propertysource


**I have added @Enablescheduling**
![image](https://user-images.githubusercontent.com/25194485/109271041-ec757b00-7849-11eb-8550-08b32d34768e.png)

**Comment by djgeary** [comment]
We are seeing a lot of our services appearing as 'applications.unknown' instead of 'up' after upgrading to Spring Boot Admin 2.4.0

**Comment by star4j** [comment]
I tried spring boot admin 2.3.1, it is no problem

**Comment by lltx** [comment]
![image](https://user-images.githubusercontent.com/50090230/109421471-6b1d1480-7a12-11eb-99a4-eed24a746b1d.png)

the same problem after upgrading to 2.4.0

**Comment by nathankooij** [comment]
We also hit this bug after upgrading to SBA 2.4.0 using Spring Cloud Kubernetes Discovery. I noticed that events got "lost" and as such, certain `AbstractEventHandler`s (like `EndpointDetectionTrigger`) were not triggered at all. Subsequently, services are reported as down or endpoints are not discovered.

I traced this back to `InstanceEventPublisher`, where after the Reactor Core upgrade to 3.4.x, the new sinks API was used. However, the new sinks API has one major difference with the previous API (`UnicastProcessor`), in the sense that serialization (i.e. concurrent access) is _not_ handled for you. Instead, the emission fails silently (adding `orThrow()` will reveal the errors).

I can fix it locally by either retrying the emission (using a `EmitFailureHandler`) or by adding `synchronized` to `InstanceEventPublisher#publish`. I can provide a PR for this if you want, @SteKoe.
